libarchive/ffi.py

```diff
@@ -125,7 +125,7 @@ def ffi(name, argtypes, restype, errcheck=None):
 ffi('entry_rdevmajor', [c_archive_entry_p], c_uint)
 ffi('entry_rdevminor', [c_archive_entry_p], c_uint)
 
-ffi('entry_set_size', [c_archive_entry_p, c_int], c_int)
+ffi('entry_set_size', [c_archive_entry_p], c_longlong)
```

Comment:
Unfortunately that's worse, it should be `ffi('entry_set_size', [c_archive_entry_p, c_longlong], c_int)`. The list represents the arguments passed to the C function. The fact that the tests didn't catch this mistake means that they're not very thorough.